pkg/server/registry/apigroups/acorn/imageallowrules/validator.go

```diff
@@ -0,0 +1,56 @@
+package imageallowrules
+
+import (
+	"context"
+
+	apiv1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+	internalv1 "github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+type ClusterValidator struct{}
+
+func (s *ClusterValidator) Validate(ctx context.Context, obj runtime.Object) (result field.ErrorList) {
+	aiar := obj.(*apiv1.ClusterImageAllowRules)
+	result = append(result, validateSignatureRules(ctx, aiar.Signatures)...)
+	return
+}
+
+func (s *ClusterValidator) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return s.Validate(ctx, obj)
+}
+
+func (s *ClusterValidator) ValidateCreate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return s.Validate(ctx, obj)
+}
+
+type ProjectValidator struct{}
+
+func (s *ProjectValidator) Validate(ctx context.Context, obj runtime.Object) (result field.ErrorList) {
+	aiar := obj.(*apiv1.ProjectImageAllowRules)
+	result = append(result, validateSignatureRules(ctx, aiar.Signatures)...)
+	return
+}
+
+func (s *ProjectValidator) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return s.Validate(ctx, obj)
+}
+
+func (s *ProjectValidator) ValidateCreate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return s.Validate(ctx, obj)
+}
+
+func validateSignatureRules(ctx context.Context, sigRules internalv1.ImageAllowRuleSignatures) (result field.ErrorList) {
+	if sigRules.Rules == nil || len(sigRules.Rules) == 0 {
```

Comment:
This can be simplified:
```suggestion
	if len(sigRules.Rules) == 0 {
```